Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XRSDK/WindowsMixedRealitySpatialMeshObserver.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.SpatialAwareness;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using UnityEngine;
+
+#if WMR_ENABLED
+using UnityEngine.XR.WindowsMR;
+#endif // WMR_ENABLED
+
+namespace Microsoft.MixedReality.Toolkit.XRSDK.WindowsMixedReality
+{
+    [MixedRealityDataProvider(
+        typeof(IMixedRealitySpatialAwarenessSystem),
+        SupportedPlatforms.WindowsUniversal,
+        "XR SDK Windows Mixed Reality Spatial Mesh Observer",
+        "Profiles/DefaultMixedRealitySpatialAwarenessMeshObserverProfile.asset",
+        "MixedRealityToolkit.SDK")]
+    [HelpURL("https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/SpatialAwareness/SpatialAwarenessGettingStarted.html")]
+    public class WindowsMixedRealitySpatialMeshObserver :
+        GenericXRSDKSpatialMeshObserver,
+        IMixedRealityCapabilityCheck
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="registrar">The <see cref="IMixedRealityServiceRegistrar"/> instance that loaded the service.</param>
+        /// <param name="name">Friendly name of the service.</param>
+        /// <param name="priority">Service priority. Used to determine order of instantiation.</param>
+        /// <param name="profile">The service's configuration profile.</param>
+        public WindowsMixedRealitySpatialMeshObserver(
+            IMixedRealitySpatialAwarenessSystem spatialAwarenessSystem,
+            string name = null,
+            uint priority = DefaultPriority,
+            BaseMixedRealityProfile profile = null) : base(spatialAwarenessSystem, name, priority, profile)
+        { }
+
+        /// <inheritdoc/>
+        protected override void ConfigureObserverVolume()
+        {
+            if (SpatialAwarenessSystem == null || XRSDKSubsystemHelpers.MeshSubsystem == null)
+            {
+                return;
+            }
+
+            // Update the observer
+            switch (ObserverVolumeType)
+            {
+                case VolumeType.AxisAlignedCube:
+                    XRSDKSubsystemHelpers.MeshSubsystem.SetBoundingVolume(ObserverOrigin, ObservationExtents);
+                    break;
+#if WMR_ENABLED
```

Comment:
Can you elaborate (I think just in this PR review, not in the code) why the sphere/useraligned cube are restricted to WMR_ENABLED but the other line isn't? Obviously I have not that much background here, it's just usually interesting to see when specific functionality in otherwise similar-looking namespaces/names is gated.

Reply:
Yep! XR SDK "core" doesn't support these additional volume types, since it supports a common cross section and WMR supports more than other platforms.
As a result, there are WMR-specific extensions in the WMR XR SDK package to allow using them. I have a custom define in this assembly definition that keys off the existence of the WMR-support package for XR SDK.